src/api/java/com/minecolonies/api/util/InventoryUtils.java

```diff
@@ -2156,12 +2156,58 @@ public static boolean tryRemoveStackFromItemHandler(final IItemHandler handler,
         return false;
     }
 
+    /**
+     * Tries to remove a group of stacks with its size from a given Itemhandler. Only removes stacks if every stack's full count is present.
+     *
+     * @param handler      the itemHandler.
+     * @param input        the stack to remove.
+     * @return true if removed the stack
+     */
+    public static boolean tryRemoveStackFromItemHandler(final IItemHandler handler, final List<ItemStorage> input)
+    {
+        List<ItemStack> removedItems = new ArrayList<>();
+        for(final ItemStorage is : input)
+        {
+            int amount = is.getItemStack().getCount();
+
+            for (int i = 0; i < handler.getSlots(); i++)
+            {
+                if (ItemStackUtils.compareItemStacksIgnoreStackSize(handler.getStackInSlot(i), is.getItemStack(), !is.ignoreDamageValue(), !is.ignoreNBT()))
+                {
+                    amount = amount - handler.extractItem(i, amount, false).getCount();
```

Comment:
Not 100% happy with all this. It's probably better to first loop over all and do a count of the inventory and if the count is fine we loop and remove. For both we have utils already. I know it's a bit less efficient but this is "on button press" logic that is executed not so often, so it should be fine

Reply:
Wasn’t sure on that myself. Want the non-List variant of the method changed as well?

Reply:
sure